doc/man/man3/suite_create.3.in

```diff
@@ -8,7 +8,7 @@
 .\" Back-Cover Texts.  A copy of the license is included in
 .\" the section entitled "GNU Free Documentation License".
 .\"
-.TH "suite_create" "3" "October 25, 2020" "0.15.2" "LibCheck"
+.TH "suite_create" "3" "@LIBCHECK_BUILD_DATE@" "@PROJECT_VERSION@" "LibCheck"
```

Comment:
Is reporting the date important?

The reason is, when someone builds the library today, and then takes the same source and builds it again tomorrow, it is worth while for the build to produce the same result. Having timestamps breaks build reproducibility.

https://reproducible-builds.org/docs/timestamps/

Reply:
Initially, the intent was to record the date of the last meaningful change to the man page. As I was making the modifications to the build system yesterday to automate the installation of the man pages, however, I figured simply setting each man page's date field to the build date was a simple heuristic.

I agree with you, though; I think the current system of using whatever date the project was configured on ultimately provides no real value or benefit. I do like the original idea of having the date of the last meaningful change to the man page, since I think it provides meaningful context to the information, the requirement that anyone modifying a man page update the date does not seem unreasonably strict nor difficult, and it would also not affect build reproducibility. What do you think?

Thanks for the link, it made a lot of great points I had not considered before.

Reply:
I'm not really sure what the date buys. There are Linux distros that ship older versions of Check. After the man pages are released there will certainly be some older versions around for what will then be older versions of Check. As long as the man pages that ship with a given version apply to that version, and the version is mentioned in the man page, I think that should be sufficient.

Can you give an example of a man page that has a date? I pulled up a few random ones for command line utilities and system calls and none of them happened to have a date. Is having a date for libraries standard practice?

One final point. If the date is in the man pages, how likely is it that new changes to the man pages will always remember to update the date line? Or, is it something that would be updated before a release?

Reply:
Really? The man pages I consulted when writing the ones for the library were `man(1)`, `man(7)`, and `man-pages(7)` and I'm pretty sure they all had dates, but I could be wrong. I'm on Arch Linux, I'm not sure if that matters. I think I remember seeing man pages on Debian/Ubuntu that the Arch Linux `man-pages` package doesn't have, so maybe that's where the difference is coming from?

 My thinking was that while updating the date would be required for all documentation changes, the primary focus would be ensuring the documentation accompanying each release is up to date, so I'm not married to the idea of including the date. I really don't dislike the idea, but the primary reason I included it in the first place was because it was specified in `man-pages(7)`. I actually agree with you; I think the library version should be enough. I'll go ahead and submit an update removing the date.

Regarding the documentation updates, I tend to agree with the OpenBSD model, where docimentation is either both complete and completely correct, or it's a bug. If a change was made that simply added to the documentation then all is fine, but if the documentation is changed to reflect new functionality or a new API but the version number or date are not updated, then the documentation is incorrect. Pull requests requesting to merge any such changes would therefore be introducing bugs, and should be fixed before being merged.

Reply:
Ooooh, the dates are at the bottom of the man pages, not the top. I see them now. Never noticed them before now. My bad.

> the primary reason I included it in the first place was because it was specified in man-pages(7)

How about this. If it turns out that having a data is necessary to get it to work, a fixed date can be put in. It would be good to have the date specified in one place, if possible, so all man pages display the same date.

> Regarding the documentation updates, I... docimentation is either both complete and completely correct, or it's a bug.

I agree with this. It would be good to enforce this, in a test that runs in GitHub Actions on PRs. Maybe there is precedent for something like that. A TODO for the future.

Reply:
Haha, no worries, I remember doing the exact same thing myself when I first read that man pages were supposed to have dates.

> If it turns out that having a data is necessary to get it to work, a fixed date can be put in.

Sure, that sounds good to me. A good example of a date being useful that I forgot when I was replying last night is SDL. SDL2 does not come with man pages but SDL does, and I only realized because the date on the man page for a function common to both library versions said it was from 2001, like ten years before SDL2 was even announced.

For now, I'll just submit an update with an empty date string, and I'll remove the Autoconf timestamp macro file, since we no longer need it.

> It would be good to enforce this, in a test that runs in GitHub Actions on PRs.

I agree, I would love to be able to test this. I love Python's [`doctest`](https://docs.python.org/3/library/doctest.html) module, and it even looks like Rust has this functionality [built into the language itself](https://doc.rust-lang.org/1.7.0/book/testing.html#documentation-tests). I'll make a note of this, and I'll start looking for ways to do this in a lightweight, cross-platform way.

Reply:
OK. Just let me know when the update with the empty date string is in.